### ISSUE TITLE
Need way to create IDataPortal instance for lazy loading

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#2658]
There is a scenario where, in a domain class property implementation, an instance of `IDataPortal<T>` is required to lazy load a property value.

Maybe the base classes need to expose a protected `GetDataPortal<T>` method? Or expose `ApplicationContext` as a protected value?

Discussed in #2646

**Comment by swegele** [comment]
@rockfordlhotka  I just sat down during my week off to give a first try to upgrade my business layer from 5.5.1 to 6.  I wanted to see how ready we are for .Net 6 Server Side Blazor.  Both of the issues you mention stopped me in my tracks immediately:

1. The inability to get ApplicationContext.  
I've got a version of ConnectionManager which keeps only one DAL connection open to the database at a time (LLBLGen ORM).  I store it the connection manager in ApplicationContext.LocalContext.  It is initialized by the first business object on server side of dataportal calling the Using statement.  Additional (child) objects can also call the Using statements, but they will get the existing version which is already connected.  It has a ref counter so that everyone gets the same connection till the first caller finally disposes at the end of its Using statement.  
Anyway, if you make it possible for the Base Classes to expose a "protected" version of their ApplicationContext then I should be good. 
It looks like you've already got the interface `IUseApplicationContext` ?

2. The inability to call DataPortal.Create/Fetch/Delete etc. from my business object static factory methods.
ALL of my business objects follow this pattern of static factory methods:

`
        public static AgencyEdit NewAgency()
        {
            var newObj = DataPortal.Create<AgencyEdit>();
            newObj.BusinessRules.CheckRules();
            return newObj;
        }
`
and
`        public static AgencyEdit GetAgency(string id)
        {
            var newObj = DataPortal.Fetch<AgencyEdit>(id);
            newObj.BusinessRules.CheckRules();
            return newObj;
        }
`

I'll be completely honest, all the DI stuff still really gets me confused.  It just doesn't click yet.  But I'd like it if I can use it or not depending on the situation.  

I guess I'm saying something lit up in me and got me excited when I read #2646 and saw that you created this ticket.  Allow me to say "Yes please!  Can you provide a way for me to get BOTH (a dataportal instance AND the ApplicationContext).  

But more than just in a Child/LazyLoad scenario...but in a static factory method (when there is no business object yet) can you give me a way to get a DataPortal like I am used to doing?

**Comment by swegele** [comment]
@rockfordlhotka Sorry, I just noticed the similarity with #2632 .  Ugh!  I'm really feeling confused and scared about this upgrade.  I mean you used the term 
> the concept "infects" all the rest of your code.

Your not kidding!  I'm looking at big changes here because DI is now REQUIRED??  You suggest refactoring the UI to use DI.  There is the rub.  How do I do that? :-)   I've used CSLA for decades, and now I'm stuck not knowing how to do the most basic things, (admittedly) because I don't yet fully understand the DI stuff and how to go "all-in" with it.  I wonder how many of us are out there.

I don't like the thought of not being able to hide/abstract away the concept of the DataPortal INSIDE my business objects.  It feels like that a fairly complex concept that is now required by consumers of the business layer.  That has been happening more and more lately (public constructors required, and public PropertyInfo objects required.)  User's come to my business objects and they are like "wow what's all this stuff", whereas before I could hide away most of it leaving a nice easy breadcrumb path for them to follow.
